[WIP] Add DefinedTerm profile

sneumann opened the pull request:
## Internal reference
<!--- If this PR fixes an open issue, please link to the issue here. -->
Fixes 
https://github.com/BioSchemas/specifications/issues/652

## Description
<!--- Describe your changes in detail. Bullet points are recommended. -->
- Adding validation, better description and examples for ontology and terminology service backed DefinedTerms

## Motivation and context
<!--- Why is this change required? What problem does it solve? -->
See https://github.com/BioSchemas/specifications/issues/652

## Have these been tested?
<!--- If applicable, please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, tests ran to see how -->
<!--- your change affects other areas of the code, etc. -->
- *e.g. New draft tested against JSON document...*

## What should reviewers focus on?
<!--- Provide suggestions for what someone should check to help with the review. -->
Since no new properties were added on top of the SchemaOrg type, only the description, marginality and cardinality need checking. 

## Types of changes
<!--- What types of changes does your PR introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New content (non-breaking change which adds new content)
- [ ] Modified content (non-breaking change which modifies existing content)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Future TO-DOs
<!--- What should be remembered to be done in the future to follow up the work in this PR -->
- [ ] Ping resources to update their DefinedTerms to this profile

sneumann committed:
Adding (empty) files for DefinedTerm profile

gtsueng reviewed (changes requested):
@sneumann The JSON Schema validation rules were missing. I have added them into the DefinedTerm jsonld file

gtsueng reviewed (changes requested):
`sameAs` should be recommended for consistency with the rest of the Bioschemas profiles -- updated in the JSON-LD file

sneumann committed:
Adding a bit of rationale why we need DefinedTerm Profile

Commit:
Augment usecase

gtsueng reviewed (changes requested):
Another potential use case for the readme:

Additionally, there may be controlled vocabularies in use that are not formalized into an ontology. In such cases, there may not be a suitable IRI and additional information would be needed. Eg- The ClinicalTrials.gov's [Protocol Registration and Results System (PRS) Schema](https://prsinfo.clinicaltrials.gov/definitions.html) has controlled vocabulary used as values for many properties but these terms are not necessarily in a formal ontology. In this case, it is important leverage the DefinedTermSet profile to ensure sufficient information is included for provenance.

gtsueng committed:
Update DefinedTerm_v0.1-DRAFT.jsonld

adding json schema validation rules

gtsueng committed:
Update DefinedTerm_v0.1-DRAFT.jsonld

added JSON Schema validation rules

gtsueng committed:
Update DefinedTerm_v0.1-DRAFT.jsonld

update JSON-schema validation

gtsueng committed:
Update DefinedTerm_v0.1-DRAFT.jsonld

update the json schema validation rules to match the marginality constraints described in https://docs.google.com/spreadsheets/d/1Cpm4-r4h39IVeQW987fgkR4byk1leXdKAjckg-0hbdU/edit#gid=0

sneumann committed:
Fix typo in DefinedTerm README

sneumann committed:
Added examples by Rene Meier and Ivan Mičetić

gtsueng committed:
Update DefinedTerm_v0.1-DRAFT.jsonld

removed extra comma that was causing an error

gtsueng committed:
Update DefinedTerm_v0.1-DRAFT.jsonld

edit namespace to minimize chance of confusing namespace with class by downstream processing scripts

gtsueng reviewed (approved):
Looks good to go to me
